keps/sig-node/24-apparmor/README.md

```diff
@@ -0,0 +1,816 @@
+
+Scenario 5 represents the case of failing the existing validation, which is
+defined at [Pod API](#pod-api).
+
+### Version Skew Strategy
+
+All API skew is resolved in the API server. New Kubelets will only use the
+AppArmor values specified in the fields, and ignore the annotations.
+
+#### Pod Creation
+
+If no AppArmor annotations or fields are specified, no action is necessary.
+
+If the `AppArmor` feature is disabled per feature gate, then the annotations and
+fields are cleared (current behavior).
+
+If the pod's OS is `windows`, fields are forbidden to be set and annotations
+are not copied to the corresponding fields.
+
+If _only_ AppArmor fields are specified, add the corresponding annotations. If these
+are specified at the Pod level, copy the annotations to each container that does
+not have annotations already specified. This ensures that the fields are enforced
+even if the node version trails the API version (see [Version Skew Strategy](##version-skew-strategy)).
+
+If _only_ AppArmor annotations are specified, copy the values into the
+corresponding fields. This ensures that existing applications continue to
+enforce AppArmor, and prevents the kubelet from needing to resolve annotations &
+fields. If the annotation is empty, then the `runtime/default` profile will be
+used by the CRI container runtime. If a localhost profile is specified, then
+container runtimes will strip the `localhost/` prefix, too. This will be covered
+by e2e tests during the GA promotion.
+
+If both AppArmor annotations _and_ fields are specified, the values MUST match.
+This will be enforced in API validation.
+
+If a Pod with a container specifies an AppArmor profile by field/annotation,
+then the container will only apply the Pod level field/annotation if none
+are set on the container level.
+
+To raise awareness of annotation usage (in case of old automation), a warning
+mechanism will be used to highlight that support will be dropped in v1.30.
+The mechanisms being considered are audit annotations, annotations on the
+object, events, or a warning as described in [KEP
+#1693](/keps/sig-api-machinery/1693-warnings).
+
+#### PodSecurityPolicy Support
+
+PodSecurityPolicy support has been removed in Kubernetes 1.25. Due to this,
+the equivalent functionality for annotations such as the below will not be supported:
+
+ * `apparmor.security.beta.kubernetes.io/defaultProfileName`
+ * `apparmor.security.beta.kubernetes.io/allowedProfileNames`
+
+To provide this functionality, users will need to implement equivalent rules in admission control,
+by injecting or validating a `LocalHostProfile` field for Pods or individual containers.
+
+#### Pod Update
+
+The AppArmor fields on a pod are immutable, which also applies to the
+[annotation](https://github.com/kubernetes/kubernetes/blob/b46612a74224b0871a97dae819f5fb3a1763d0b9/pkg/apis/core/validation/validation.go#L177-L182).
+
+When an [Ephemeral Container](20190212-ephemeral-containers.md) is added, it
```

Comment:
ephemeral containers are GA now: https://github.com/kubernetes/enhancements/issues/277

The tests section doesn't mention e2e tests validating ephemeral containers behavior. Should this be added?